osu.Game.Tests/Visual/Online/TestSceneChatOverlay.cs

```diff
@@ -576,6 +582,76 @@ public void TestTextBoxSavePerChannel()
             });
         }
 
+        [Test]
+        public void TestChatReport()
+        {
+            ChatReportRequest request = null;
+
+            AddStep("Show overlay with channel", () =>
+            {
+                chatOverlay.Show();
+                channelManager.CurrentChannel.Value = channelManager.JoinChannel(testChannel1);
+            });
+
+            AddAssert("Overlay is visible", () => chatOverlay.State.Value == Visibility.Visible);
+            waitForChannel1Visible();
+
+            AddStep("Setup request handling", () =>
+            {
+                requestLock.Reset();
+
+                dummyAPI.HandleRequest = r =>
+                {
+                    if (!(r is ChatReportRequest req))
+                        return false;
+
+                    Task.Run(() =>
+                    {
+                        request = req;
+                        requestLock.Wait(10000);
+                        req.TriggerSuccess();
+                    });
+
+                    return true;
+                };
+            });
+
+            AddStep("Show report popover", () => this.ChildrenOfType<DrawableUsername>().First().ShowPopover());
+
+            AddStep("Set report reason to other", () =>
+            {
+                var reason = this.ChildrenOfType<OsuEnumDropdown<ChatReportReason>>().Single();
+                reason.Current.Value = ChatReportReason.Other;
+            });
+
+            AddStep("Try to report", () =>
+            {
+                var btn = this.ChildrenOfType<ReportChatPopover>().Single().ChildrenOfType<RoundedButton>().Single();
+                InputManager.MoveMouseTo(btn);
+                InputManager.Click(MouseButton.Left);
+            });
+
+            AddWaitStep("Wait", 3);
```

Comment:
Wait steps cannot be used like this. There's no guarantee of timing. What are you waiting for? You probably want to use an `UntilStep` with a condition instead.